### PR TITLE
Fix windows batch script to pre java 9 and java 9

### PR DESCRIPTION
**Pull request by neomatrix369**
On the back of the conversation about Windows batch script failing for the recording program.

Added echoes displaying system info and important environment variables and removed reserved characters like `>` and `<` from echoes. <=== these were causing the error '..specified file cannot be found'

Also fixed two usages of the set variable statements for `JAVA_HOME` and `JAVA_FULL_VERSION` <=== this is what the user had as an issue and the other implementation of `set` it `set "VARIABLE=VALUE"` surrounded by quotes.

Here's a snapshot of the run logs:
```

Displaying operating system specific systeminfo...
OS Name:                   Microsoft Windows 10 Enterprise Evaluation
OS Version:                10.0.15063 N/A Build 15063
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
BIOS Version:              innotek GmbH VirtualBox, 12/1/2006

JAVA_FULL_VERSION=9.0.1

JAVA_VERSION=9
"--- Java 9 or higher detected (Java version 9) ---"
"Adding JVM args to the DEFAULT_JVM_OPTS variable, new value set to ''"
--------------------------------------------------------------------------------------------------------------

C:\Users\IEUser\Downloads\tdl-lord-of-runners\template-for-recording>"java.exe" --illegal-access=warn --add-modules=java.xml.bind,java.activation    -jar "C:\Users\IEUser\Downloads\tdl-lord-of-runners\template-for-recording\\record\record-and-upload-capsule.jar" --config C:\Users\IEUser\Downloads\tdl-lord-of-runners\template-for-recording\\config\credentials.config --store C:\Users\IEUser\Downloads\tdl-lord-of-runners\template-for-recording\\record\localstore --sourcecode C:\Users\IEUser\Downloads\tdl-lord-of-runners\template-for-recording\
WARNING: Illegal reflective access by Capsule (file:/C:/Users/IEUser/Downloads/tdl-lord-of-runners/template-for-recording/record/record-and-upload-capsule.jar) to field com.sun.jmx.mbeanserver.JmxMBeanServer.mbsInterceptor
INFO  [main]       - Starting recording app
INFO  [main]       - Checking permissions
ERROR [main]       - User does not have enough permissions to upload. Reason: The provided token has expired. (Service: Amazon S3; Status Code: 400; Error Code: ExpiredToken; Request ID: 012B90623F1A3A91)
```

### CODE REVIEW

**Comment by neomatrix369** [comment]
I agree some of the echos added are already taken care of by the DEBUG switch but I suggest its best to display these echos at all times, given the issues we have had.

**Comment by neomatrix369** [comment]
@julianghionoiu could you also please run this script on Windows 10 before we merge the PR

File: template-for-recording/record_screen_and_upload.bat [hunk 1]
```diff
@@ -32,9 +36,15 @@ echo location of your Java installation.
 goto fail
 
 :findJavaFromJavaHome
-set JAVA_HOME=%JAVA_HOME:"=%
+set "JAVA_HOME=%JAVA_HOME:"=%
```
Comment (neomatrix369) [inline template-for-recording/record_screen_and_upload.bat]: You mentioned this and we also learnt from our users' fix

File: template-for-recording/record_screen_and_upload.bat [hunk 2]
```diff
@@ -77,24 +87,32 @@ set PARAM_SOURCECODE_DIR=--sourcecode %APP_HOME%
 for /f "tokens=3" %%g in ('%JAVA_EXE% -version 2^>^&1 ^| findstr /i "version"') do (
     set JAVA_FULL_VERSION=%%g
 )
-set JAVA_FULL_VERSION=%JAVA_FULL_VERSION:"=%
+set "JAVA_FULL_VERSION=%JAVA_FULL_VERSION:"=%
```
Comment (neomatrix369) [inline template-for-recording/record_screen_and_upload.bat]: Same here

File: template-for-recording/record_screen_and_upload.bat (hunk 1 above)
Comment (julianghionoiu) [reply template-for-recording/record_screen_and_upload.bat]: I do not think this is the problem. `JAVA_HOME=%JAVA_HOME:"=%` is a valid substitution syntax.

File: template-for-recording/record_screen_and_upload.bat (hunk 2 above)
Comment (julianghionoiu) [reply template-for-recording/record_screen_and_upload.bat]: We should not change the substitution syntax.
I think the problem is that %JAVA_FULL_VERSION% ends up not being defined.
Try this statement without defining JAVA_FULL_VERSION.

File: template-for-recording/record_screen_and_upload.bat [hunk 3]
```diff
@@ -32,9 +36,15 @@ echo location of your Java installation.
 goto fail
 
 :findJavaFromJavaHome
-set JAVA_HOME=%JAVA_HOME:"=%
+set "JAVA_HOME=%JAVA_HOME:"=%
 set JAVA_EXE=%JAVA_HOME%/bin/java.exe
```
Comment (julianghionoiu) [inline template-for-recording/record_screen_and_upload.bat]: This is where the odd slashes are coming from. I think the path should add backslashes.

**Comment by julianghionoiu** [comment]
@neomatrix369 which test cases have you covered?

File: template-for-recording/record_screen_and_upload.bat (hunk 1 above)
Comment (julianghionoiu) [reply template-for-recording/record_screen_and_upload.bat]: Also notice the `if defined JAVA_HOME goto findJavaFromJavaHome` on line 24. This ensures that the variable is defined.

File: template-for-recording/record_screen_and_upload.bat (hunk 3 above)
Comment (neomatrix369) [reply template-for-recording/record_screen_and_upload.bat]: Changed now - maybe the first time it's going thru this path since the script was written

**Comment by neomatrix369** [comment]
@neomatrix369 both Java 9 and pre-Java 9 tests, also did a test where `JAVA_HOME` exists and this path which might not have been traversed previously might be an issue to investigate, although a separate thread of investigation.

**Comment by neomatrix369** [comment]
Now also do specific edge case tests with `JAVA_FULL_VERSION` and wasn't able to reproduce the reported issue. The error "The syntax of the command is incorrect." could occur if a command is not supplied with the right parameters and e.g. just calling `mv` or `rm` or `copy`.

File: template-for-recording/record_screen_and_upload.bat (hunk 2 above)
Comment (neomatrix369) [reply template-for-recording/record_screen_and_upload.bat]: I have tested with the not defined scenario and it works fine. And likewise, the syntax is similar to 
the JAVA_HOME one previously, which leaves us with a red-herring.

**Review by julianghionoiu — APPROVED** [review]
